RIOT-OS/RIOT
# gnrc_dhcpv6_client_mud_url: rename to dhcpv6_client_mud_url

### Contribution description
There is no real reason for that pseudo-module to use the `gnrc_` prefix. Neither does it need GNRC-components (except, but optionally, as a network stack of course), nor is it implemented within the GNRC network stack.



### Testing procedure
`tests/gnrc_dhcpv6_client_6lbr` should not change when build in this PR and on the merge base to master. The test still passes.

```console
$ RIOT_CI_BUILD=1 make -C tests/gnrc_dhcpv6_client_6lbr/ flash -j test-as-root
```



### Issues/PRs references
Follow-up on https://github.com/RIOT-OS/RIOT/pull/15508
